app/src/main/java/com/mealsmadeeasy/ui/login/LoginActivity.kt

```diff
@@ -0,0 +1,92 @@
+package com.mealsmadeeasy.ui.login
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.support.design.widget.Snackbar
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.SignInButton
+import com.google.android.gms.common.api.ApiException
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.GoogleAuthProvider
+import com.mealsmadeeasy.R
+import com.mealsmadeeasy.ui.BaseActivity
+import com.mealsmadeeasy.ui.home.HomeActivity
+
+class LoginActivity : BaseActivity() {
+
+    private val RC_SIGN_IN = 9001
+
+    private lateinit var auth: FirebaseAuth
+    private lateinit var signInClient: GoogleSignInClient
+
+    companion object {
+        fun newIntent(context: Context) = Intent(context, LoginActivity::class.java)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+        findViewById<SignInButton>(R.id.sign_in_button).setOnClickListener{signIn()}
+
+        // Configure sign-in to request the user's ID, email address, and basic
+        // profile. ID and basic profile are included in DEFAULT_SIGN_IN.
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(getString(R.string.default_web_client_id)) //OAuth 2.0 client ID
+                .requestEmail()
+                .build()
+
+        // Build a GoogleSignInClient with the options specified by gso.
+        signInClient = GoogleSignIn.getClient(this, gso)
+        auth = FirebaseAuth.getInstance()
+    }
+
+    public override fun onStart() {
+        super.onStart()
+        // Check if user is signed in (non-null) and update UI accordingly.
+//        val currentUser = auth.currentUser
+//        if (currentUser != null) {
+//            startActivity(HomeActivity.newIntent(this))
+//        }
```

Comment:
We should be able to kill the entire method here since it's just calling `super`.